apps/src/dance/lab2/ProgramExecutor.ts

```diff
@@ -79,7 +79,7 @@ export default class ProgramExecutor {
 
     this.hooks = await this.compileAllCode(code);
     if (!this.hooks.runUserSetup || !this.hooks.getCueList) {
-      Lab2MetricsReporter.logWarning('Missing required hooks in compiled code');
+      this.reportMissingHooks('runUserSetup', 'getCueList');
       return;
     }
```

Comment:
Worth splitting into two separate checks?

Reply:
Hmm we could, but both of these are necessary in conjunction to start code execution

Reply:
Oh yes, we'd have to return in either case.